src/bases/rt2space.jl

```diff
@@ -0,0 +1,133 @@
+        if cellpairs[2,i] > 0
+            c1 = cellpairs[1,i]; # cell1 = Cells[c1] #mesh.faces[c1]
+            c2 = cellpairs[2,i]; # cell2 = Cells[c2] #mesh.faces[c2]
+
+            cell1 = CompScienceMeshes.indices(mesh, c1)
+            cell2 = CompScienceMeshes.indices(mesh, c2)
+
+            e1, e2 = getcommonedge(cell1, cell2)
+            functions[2*i-1] = [
+              Shape{T}(c1, 2*abs(e1)+chooseref1(sign(e1)), T(+1.0)),
+              Shape{T}(c2, 2*abs(e2)+chooseref1(sign(e2)), T(-1.0))]
+            functions[2*i] = [
+              Shape{T}(c1, 2*abs(e1)+chooseref2(sign(e1)), T(+1.0)),
+              Shape{T}(c2, 2*abs(e2)+chooseref2(sign(e2)), T(-1.0))]  
+            isct = intersect(cell1, cell2)
+            @assert length(isct) == 2
+            @assert !(cell1[abs(e1)] in isct)
+            @assert !(cell2[abs(e2)] in isct)
+
+            ctr1 = cartesian(center(chart(mesh, c1)))
+            ctr2 = cartesian(center(chart(mesh, c2)))
+            positions[2*i-1] = (ctr1 + ctr2) / 2
+            positions[2*i] = (ctr1 + ctr2) / 2
+        else
+            c1 = cellpairs[1,i]
+            e1 = cellpairs[2,i]
+            functions[i] = [
+            Shape(c1, abs(e1), T(+1.0))]
+            positions[i] = cartesian(center(chart(mesh, c1)))
+        end
+    end
+
+    for (i,cell) in enumerate(Cells)
+        functions[2*numpairs+2*i-1] = [
+              Shape{T}(i, 7, T(+1.0))]
+        
+        functions[2*numpairs+2*i] = [
+              Shape{T}(i, 8, T(+1.0))]
+
+        ctr1 = cartesian(center(chart(mesh, i)))
+        positions[2*numpairs+2*i-1] = ctr1
+        positions[2*numpairs+2*i] = ctr1
+    end
+
+    geo = mesh
+    RT2Basis(geo, functions, positions)
+end
+
+
+function raviartthomas2(mesh, edges::CompScienceMeshes.AbstractMesh{U,2} where {U})
+    cps = CompScienceMeshes.cellpairs(mesh, edges)
+    # ids = findall(x -> x>0, cps[2,:])
+    raviartthomas2(mesh, cps)
+end
+
+function raviartthomas2(mesh::CompScienceMeshes.AbstractMesh{U,3} where {U})
+    bnd = boundary(mesh)
+    edges = submesh(!in(bnd), skeleton(mesh,1))
+    return raviartthomas2(mesh, edges)
+end
+
+
+"""
+    raviartthomas(mesh)
+
+Conducts pre-processing on the input `mesh` by extracting the cell edges, cell pairs
+    and indices required to construct the RT basis on the `mesh`.
+
+Calls raviartthomas(mesh::Mesh, cellpairs::Array{Int,2}), which constructs
+    the RT basis on the `mesh`, using the cell pairs identified.
+
+Returns the RT basis object.
+"""
+function raviartthomas2(mesh; sort=:spacefillingcurve)
+    edges = skeleton(mesh, 1; sort)
+    cps = cellpairs(mesh, edges, dropjunctionpair=true)
+    ids = findall(x -> x>0, cps[2,:])
+    raviartthomas2(mesh, cps[:,ids])
+end
+
+divergence(X::RT2Basis, geo, fns) = LagrangeBasis{0,-1,1}(geo, fns, deepcopy(positions(X)))
+ntrace(X::RT2Basis, geo, fns) = LagrangeBasis{0,-1,1}(geo, fns, deepcopy(positions(X)))
+
+
+function LinearAlgebra.cross(::NormalVector, s::RT2Basis)
```

Comment:
This does not make sense I think for the RT2 basis. Its rotated version should be the Nd2 space, not the Nd(1) space.

Reply:
Yes. I am commenting this part for now but will revise it when I push the ND2Basis.